engine/src/flutter/shell/platform/android/io/flutter/plugin/view/SensitiveContentPlugin.java

```diff
@@ -0,0 +1,116 @@
+  private final Activity mFlutterActivity;
+  private final SensitiveContentChannel mSensitiveContentChannel;
+
+  public SensitiveContentPlugin(
+      @NonNull int flutterViewId,
+      @NonNull Activity activity,
+      @NonNull SensitiveContentChannel sensitiveContentChannel) {
+    mFlutterActivity = activity;
+    mFlutterViewId = flutterViewId;
+    mSensitiveContentChannel = sensitiveContentChannel;
+
+    mSensitiveContentChannel.setSensitiveContentMethodHandler(this);
+  }
+
+  /**
+   * Sets content sensitivity level of the Android {@code View} associated with this plugin's {@code
+   * mFlutterViewId} to the level specified by {@requestedContentSensitivity}.
+   */
+  @Override
+  public void setContentSensitivity(
+      @NonNull int requestedContentSensitivity, @NonNull MethodChannel.Result result) {
+    if (!isSupported()) {
+      // This feature is only available on >= API 35.
+      result.error(
+          "error",
+          "Setting content sensitivity with value "
+              + requestedContentSensitivity
+              + " failed because this feature is only available on Android API >= 35.",
+          null);
+      return;
+    }
+
+    final View flutterView = mFlutterActivity.findViewById(mFlutterViewId);
+    final int initialContentSensitivity = flutterView.getContentSensitivity();
+
+    if (initialContentSensitivity == requestedContentSensitivity) {
+      // Content sensitivity for the requested View already set to requestedContentSensitivity.
+      result.success(null);
+      return;
+    }
+
+    // Set requestedContentSensitivity on the View.
+    flutterView.setContentSensitivity(requestedContentSensitivity);
+
+    // Invalidate the View to force a redraw in order to ensure that it updates to be
+    // obscured/unobscured as expected.
+    flutterView.invalidate();
+
+    result.success(null);
+  }
+
+  /**
+   * Gets content sensitivity level of the Android {@code View} associated with this plugin's {@code
+   * mFlutterViewId}.
+   */
+  @Override
+  public Integer getContentSensitivity() {
+    if (!isSupported()) {
+      // This feature is only available on >= API 35.
+      return null;
+    }
+
+    final View flutterView = mFlutterActivity.findViewById(mFlutterViewId);
+    final int currentContentSensitivity = flutterView.getContentSensitivity();
+    return currentContentSensitivity;
+  }
+
+  /**
+   * Returns whether or not marking content sensitivity is supported on the device.
+   *
+   * <p>It is supported on devices running Android API >= 35.
+   */
+  @Override
+  public void isSupported(@NonNull MethodChannel.Result result) {
+    result.success(isSupported());
+  }
+
+  private boolean isSupported() {
+    return Build.VERSION.SDK_INT >= API_LEVELS.API_35;
+  }
+
+  /**
+   * Releases all resources held by this {@code SensitiveContentPlugin}.
+   *
+   * <p>Do not invoke any methods on a {@code SensitiveContentPlugin} after invoking this method.
+   */
+  public void destroy() {
```

Comment:
I also think we need to null out the reference to mFlutterActivity so that this plugin does not hold a reference to the activity that probably cause the destroy.